`logs/pkgs/scikit_build_core.log` is
```
Using cached file /Users/dcoudert/sage/upstream/scikit_build_core-0.12.2-py3-none-any.whl
Setting up build directory /Users/dcoudert/sage/local/var/tmp/sage/build/scikit_build_core-0.12.2
No stamp file for package 'scikit_build_core' in /Users/dcoudert/sage/local/var/lib/sage/installed
No spkg-legacy-uninstall script; nothing to do
[spkg-install] Staged wheel file, staged /Users/dcoudert/sage/local/var/lib/sage/scripts/scikit_build_core/spkg-requirements.txt
Moving package files from temporary location /Users/dcoudert/sage/local/var/tmp/sage/build/scikit_build_core-0.12.2/inst to /Users/dcoudert/sage/local
[spkg-pipinst] Using pip 26.0.1 from /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/pip (python 3.13)
[spkg-pipinst] Looking in links: /Users/dcoudert/sage/local/var/lib/sage/wheels
[spkg-pipinst] Processing /Users/dcoudert/sage/local/var/lib/sage/wheels/scikit_build_core-0.12.2-py3-none-any.whl (from -r /Users/dcoudert/sage/local/var/lib/sage/scripts/scikit_build_core/spkg-requirements.txt (line 1))
[spkg-pipinst] Installing collected packages: scikit_build_core
[spkg-pipinst] Successfully installed scikit_build_core-0.12.2
Deleting build directory /Users/dcoudert/sage/local/var/tmp/sage/build/scikit_build_core-0.12.2
Finished installing scikit_build_core-0.12.2
```
and the full error message for `pyzmq`
```
[pyzmq-27.1.0] Using cached file /Users/dcoudert/sage/upstream/pyzmq-27.1.0.tar.gz
[pyzmq-27.1.0] Setting up build directory /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/var/tmp/sage/build/pyzmq-27.1.0
[pyzmq-27.1.0] No patch files found in ../patches
[pyzmq-27.1.0] Host system: Darwin MAC-04017247 25.4.0 Darwin Kernel Version 25.4.0: Thu Mar 19 19:30:44 PDT 2026; root:xnu-12377.101.15~1/RELEASE_ARM64_T6000 arm64
[pyzmq-27.1.0] C compiler: gcc -std=gnu23, Apple clang version 21.0.0 (clang-2100.0.123.102), Target: arm64-apple-darwin25.4.0, Thread model: posix, InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
[pyzmq-27.1.0] No stamp file for package 'pyzmq' in /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/var/lib/sage/installed
[pyzmq-27.1.0] No spkg-legacy-uninstall script; nothing to do
[pyzmq-27.1.0] [spkg-install] Installing pyzmq-27.1.0
[pyzmq-27.1.0] [spkg-install] * Creating isolated environment: venv+pip...
[pyzmq-27.1.0] [spkg-install] * Installing packages in isolated environment:
[pyzmq-27.1.0] [spkg-install]   - cffi; implementation_name == 'pypy'
[pyzmq-27.1.0] [spkg-install]   - cython>=3.0.0; implementation_name != 'pypy'
[pyzmq-27.1.0] [spkg-install]   - packaging
[pyzmq-27.1.0] [spkg-install]   - scikit-build-core>=0.10
[pyzmq-27.1.0] [spkg-install] > /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/bin/python3 -m pip
[pyzmq-27.1.0] [spkg-install]   --python /private/var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-
[pyzmq-27.1.0] [spkg-install]   env-z20eh9ju/bin/python install --use-pep517 --no-warn-script-location --no-
[pyzmq-27.1.0] [spkg-install]   compile -r /var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-
[pyzmq-27.1.0] [spkg-install]   requirements-0sv_do2y.txt
[pyzmq-27.1.0] [spkg-install] < Looking in links: /Users/dcoudert/sage/local/var/lib/sage/venv-
[pyzmq-27.1.0] [spkg-install]   python3.13/var/lib/sage/wheels
[pyzmq-27.1.0] [spkg-install] < Ignoring cffi: markers 'implementation_name == "pypy"' don't match your
[pyzmq-27.1.0] [spkg-install]   environment
[pyzmq-27.1.0] [spkg-install] < Processing /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/var/lib/s
[pyzmq-27.1.0] [spkg-install]   age/wheels/cython-3.2.4-cp313-cp313-macosx_26_0_arm64.whl (from -r
[pyzmq-27.1.0] [spkg-install]   /var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-
[pyzmq-27.1.0] [spkg-install]   requirements-0sv_do2y.txt (line 1))
[pyzmq-27.1.0] [spkg-install] < Processing /Users/dcoudert/sage/local/var/lib/sage/venv-
[pyzmq-27.1.0] [spkg-install]   python3.13/var/lib/sage/wheels/packaging-26.0-py3-none-any.whl (from -r
[pyzmq-27.1.0] [spkg-install]   /var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-
[pyzmq-27.1.0] [spkg-install]   requirements-0sv_do2y.txt (line 3))
[pyzmq-27.1.0] [spkg-install] < ERROR: Could not find a version that satisfies the requirement scikit-build-
[pyzmq-27.1.0] [spkg-install]   core>=0.10 (from versions: none)
[pyzmq-27.1.0] [spkg-install] < ERROR: No matching distribution found for scikit-build-core>=0.10
[pyzmq-27.1.0] [spkg-install] 
[pyzmq-27.1.0] [spkg-install] Traceback (most recent call last):
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/__main__.py", line 203, in _handle_build_error
[pyzmq-27.1.0] [spkg-install]     yield
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/__main__.py", line 601, in main
[pyzmq-27.1.0] [spkg-install]     built = build(
[pyzmq-27.1.0] [spkg-install]         args.srcdir,
[pyzmq-27.1.0] [spkg-install]     ...<5 lines>...
[pyzmq-27.1.0] [spkg-install]         installer=args.installer,
[pyzmq-27.1.0] [spkg-install]     )
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/__main__.py", line 264, in build_package
[pyzmq-27.1.0] [spkg-install]     out = _build(
[pyzmq-27.1.0] [spkg-install]         isolation,
[pyzmq-27.1.0] [spkg-install]     ...<6 lines>...
[pyzmq-27.1.0] [spkg-install]         installer,
[pyzmq-27.1.0] [spkg-install]     )
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/__main__.py", line 187, in _build
[pyzmq-27.1.0] [spkg-install]     with _bootstrap_build_env(
[pyzmq-27.1.0] [spkg-install]          ~~~~~~~~~~~~~~~~~~~~^
[pyzmq-27.1.0] [spkg-install]         isolation,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     ...<6 lines>...
[pyzmq-27.1.0] [spkg-install]         pyproject_hooks.quiet_subprocess_runner if _ctx.verbosity < 0 else None,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     ) as builder:
[pyzmq-27.1.0] [spkg-install]     ^
[pyzmq-27.1.0] [spkg-install]   File "/opt/homebrew/Cellar/python@3.13/3.13.13/Frameworks/Python.framework/Versions/3.13/lib/python3.13/contextlib.py", line 141, in __enter__
[pyzmq-27.1.0] [spkg-install]     return next(self.gen)
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/__main__.py", line 153, in _bootstrap_build_env
[pyzmq-27.1.0] [spkg-install]     install(builder.build_system_requires)
[pyzmq-27.1.0] [spkg-install]     ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/env.py", line 148, in install
[pyzmq-27.1.0] [spkg-install]     self._env_backend.install_dependencies(requirements, constraints)
[pyzmq-27.1.0] [spkg-install]     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/env.py", line 309, in install_dependencies
[pyzmq-27.1.0] [spkg-install]     run_subprocess(cmd)
[pyzmq-27.1.0] [spkg-install]     ~~~~~~~~~~~~~~^^^^^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/build/_ctx.py", line 71, in run_subprocess
[pyzmq-27.1.0] [spkg-install]     subprocess.run(cmd, capture_output=True, check=True, cwd=cwd, env=env)
[pyzmq-27.1.0] [spkg-install]     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]   File "/opt/homebrew/Cellar/python@3.13/3.13.13/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 577, in run
[pyzmq-27.1.0] [spkg-install]     raise CalledProcessError(retcode, process.args,
[pyzmq-27.1.0] [spkg-install]                              output=stdout, stderr=stderr)
[pyzmq-27.1.0] [spkg-install] subprocess.CalledProcessError: Command '['/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/bin/python3', '-m', 'pip', '--python', '/private/var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-env-z20eh9ju/bin/python', 'install', '--use-pep517', '--no-warn-script-location', '--no-compile', '-r', '/var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-requirements-0sv_do2y.txt']' returned non-zero exit status 1.
[pyzmq-27.1.0] [spkg-install] 
[pyzmq-27.1.0] [spkg-install] ERROR Command '['/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/bin/python3', '-m', 'pip', '--python', '/private/var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-env-z20eh9ju/bin/python', 'install', '--use-pep517', '--no-warn-script-location', '--no-compile', '-r', '/var/folders/hw/w6ldqpgn52vdvv03cgbkkr3h3lhtdp/T/build-requirements-0sv_do2y.txt']' returned non-zero exit status 1.
[pyzmq-27.1.0] [spkg-install] Warning: building with "python3 -m build --wheel --outdir=dist   ." failed.
[pyzmq-27.1.0] [spkg-install] Retrying with "python3 -m build --wheel --outdir=dist --no-isolation --skip-dependency-check  .".
[pyzmq-27.1.0] [spkg-install] * Building wheel...
[pyzmq-27.1.0] [spkg-install] *** scikit-build-core 0.12.2 using CMake 4.3.1 (wheel)
[pyzmq-27.1.0] [spkg-install] *** Configuring CMake...
[pyzmq-27.1.0] [spkg-install] Traceback (most recent call last):
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
[pyzmq-27.1.0] [spkg-install]     main()
[pyzmq-27.1.0] [spkg-install]     ~~~~^^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in main
[pyzmq-27.1.0] [spkg-install]     json_out["return_val"] = hook(**hook_input["kwargs"])
[pyzmq-27.1.0] [spkg-install]                              ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
[pyzmq-27.1.0] [spkg-install]     return _build_backend().build_wheel(
[pyzmq-27.1.0] [spkg-install]            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
[pyzmq-27.1.0] [spkg-install]         wheel_directory, config_settings, metadata_directory
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     )
[pyzmq-27.1.0] [spkg-install]     ^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/scikit_build_core/build/__init__.py", line 33, in build_wheel
[pyzmq-27.1.0] [spkg-install]     return _build_wheel_impl(
[pyzmq-27.1.0] [spkg-install]            ~~~~~~~~~~~~~~~~~^
[pyzmq-27.1.0] [spkg-install]         wheel_directory,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     ...<2 lines>...
[pyzmq-27.1.0] [spkg-install]         editable=False,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     ).wheel_filename
[pyzmq-27.1.0] [spkg-install]     ^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/scikit_build_core/build/wheel.py", line 177, in _build_wheel_impl
[pyzmq-27.1.0] [spkg-install]     return _build_wheel_impl_impl(
[pyzmq-27.1.0] [spkg-install]         wheel_directory,
[pyzmq-27.1.0] [spkg-install]     ...<5 lines>...
[pyzmq-27.1.0] [spkg-install]         pyproject=pyproject,
[pyzmq-27.1.0] [spkg-install]     )
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/scikit_build_core/build/wheel.py", line 434, in _build_wheel_impl_impl
[pyzmq-27.1.0] [spkg-install]     builder.configure(
[pyzmq-27.1.0] [spkg-install]     ~~~~~~~~~~~~~~~~~^
[pyzmq-27.1.0] [spkg-install]         defines=defines,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     ...<2 lines>...
[pyzmq-27.1.0] [spkg-install]         version=metadata.version,
[pyzmq-27.1.0] [spkg-install]         ^^^^^^^^^^^^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install]     )
[pyzmq-27.1.0] [spkg-install]     ^
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/scikit_build_core/builder/builder.py", line 234, in configure
[pyzmq-27.1.0] [spkg-install]     numpy_include_dir = get_numpy_include_dir()
[pyzmq-27.1.0] [spkg-install]   File "/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/lib/python3.13/site-packages/scikit_build_core/builder/sysconfig.py", line 188, in get_numpy_include_dir
[pyzmq-27.1.0] [spkg-install]     return Path(np.get_include())
[pyzmq-27.1.0] [spkg-install]                 ^^^^^^^^^^^^^^
[pyzmq-27.1.0] [spkg-install] AttributeError: module 'numpy' has no attribute 'get_include'
[pyzmq-27.1.0] [spkg-install] 
[pyzmq-27.1.0] [spkg-install] ERROR Backend subprocess exited when trying to invoke build_wheel
[pyzmq-27.1.0] [spkg-install] ********************************************************************************
[pyzmq-27.1.0] [spkg-install] Error building a wheel for pyzmq-27.1.0
[pyzmq-27.1.0] [spkg-install] ********************************************************************************
[pyzmq-27.1.0] ************************************************************************
[pyzmq-27.1.0] Error installing package pyzmq-27.1.0
[pyzmq-27.1.0] ************************************************************************
[pyzmq-27.1.0] Please email sage-devel (http://groups.google.com/group/sage-devel)
[pyzmq-27.1.0] explaining the problem and including the log files
[pyzmq-27.1.0]   /Users/dcoudert/sage/logs/pkgs/pyzmq-27.1.0.log
[pyzmq-27.1.0] and
[pyzmq-27.1.0]   /Users/dcoudert/sage/config.log
[pyzmq-27.1.0] Describe your computer, operating system, etc.
[pyzmq-27.1.0] If you want to try to fix the problem yourself, *don't* just cd to
[pyzmq-27.1.0] /Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/var/tmp/sage/build/pyzmq-27.1.0 and type 'make' or whatever is appropriate.
[pyzmq-27.1.0] Instead, the following commands setup all environment variables
[pyzmq-27.1.0] correctly and load a subshell for you to debug the error:
[pyzmq-27.1.0]   (cd '/Users/dcoudert/sage/local/var/lib/sage/venv-python3.13/var/tmp/sage/build/pyzmq-27.1.0' && '/Users/dcoudert/sage/sage' --buildsh)
[pyzmq-27.1.0] When you are done debugging, you can type "exit" to leave the subshell.
[pyzmq-27.1.0] ************************************************************************
``

--- a/build/pkgs/pyzmq/dependencies
+++ b/build/pkgs/pyzmq/dependencies
@@ -1,4 +1,4 @@
-zeromq cffi | $(PYTHON_TOOLCHAIN) $(PYTHON) cython scikit_build_core
+numpy zeromq cffi | $(PYTHON_TOOLCHAIN) $(PYTHON) cython scikit_build_core
 
 ----------
 All lines of this file are ignored except the first.

